homeassistant/components/shelly/entity.py

```diff
@@ -320,3 +375,74 @@ def device_state_attributes(self) -> dict:
             return None
 
         return self.description.device_state_attributes(self.wrapper.device.status)
+
+
+class ShellySleepingBlockAttributeEntity(ShellyBlockAttributeEntity, RestoreEntity):
+    """Represent a shelly sleeping block attribute entity."""
+
+    # pylint: disable=super-init-not-called
+    def __init__(
+        self,
+        wrapper: ShellyDeviceWrapper,
+        block: aioshelly.Block,
+        attribute: str,
+        description: BlockAttributeDescription,
+        entry: Optional[ConfigEntry] = None,
+    ) -> None:
+        """Initialize the sleeping sensor."""
+        self.last_state = None
+        self.wrapper = wrapper
+        self.attribute = attribute
+        self.block = block
+        self.description = description
+        self._unit = self.description.unit
+
+        if block is not None:
+            if callable(self._unit):
+                self._unit = self._unit(block.info(attribute))
+
+            self._unique_id = f"{self.wrapper.mac}-{block.description}-{attribute}"
+            self._name = get_entity_name(
+                self.wrapper.device, block, self.description.name
+            )
+        else:
+            self._unique_id = entry.unique_id
+            self._name = entry.original_name
+
+    @property
+    def attribute_value(self):
+        """Value of sensor."""
+        if self.block is not None:
+            return super().attribute_value
+
+        return self.last_state
```

Comment:
This needs to be a boolean value, which it's not with current `set_last_state` implementation.